sentry-core/src/test/java/io/sentry/core/UncaughtExceptionHandlerIntegrationTest.kt

```diff
@@ -0,0 +1,47 @@
+package io.sentry.core
+
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyZeroInteractions
+import com.nhaarman.mockitokotlin2.whenever
+import kotlin.test.Test
+
+class UncaughtExceptionHandlerIntegrationTest {
+    @Test
+    fun `when UncaughtExceptionHandlerIntegration is initialized, uncaught handler is unchanged`()
+    {
+        val handlerMock = mock<UncaughtExceptionHandler>()
+        UncaughtExceptionHandlerIntegration(handlerMock);
+        verifyZeroInteractions(handlerMock)
```

Comment:
Can't you test this by setting a mock `Thread.UncaughtExceptionHandler` with `Thread.setDefaultUncaughtExceptionHandler` before creating an instance of `UncaughtExceptionHandlerIntegration ` and verifying that `Thread.getDefaultUncaughtExceptionHandler()` instance equals your mock (actually I would use `assertSame` here)?

Reply:
Won't work when tests run in parallel. We shouldn't rely on any static state for unit tests

Reply:
agreed